agent/consul/leader_connect_ca_test.go

```diff
@@ -431,6 +436,7 @@ func generateCertPEM(t *testing.T, caPrivKey *rsa.PrivateKey, notBefore time.Tim
 		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
 		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
 		BasicConstraintsValid: true,
+		URIs:                  []*url.URL{connect.SpiffeIDAgent{Host: "foo"}.URI()},
```

Comment:
Confirming this change is intentional?

Reply:
Oh ya, great question! I should have given more of an explanation of this part of the change.

I was running into a panic in this test because of this line:
https://github.com/hashicorp/consul/blob/1a7b89c6cbb82ab497963846ce7e8fa39ca3b53a/agent/consul/leader_connect_ca.go#L1532

Basically the PEM encoded string coming out did not have any URIs. I tracked it back to how our fake was using the wrong certs for the local intermediate cert. Previously we would return the root cert as the "intermediate cert", and the root cert happened to have URIs on it, so the test was passing.

So once I fixed the fake to return the correct cert, I had to fix the generation of that cert to include the URIs.

The fake is still a bit strange. Normally the signed leaf cert should be the one with URIs on it, but the fake is returning an empty string for the actual leaf cert, so we have to rely on the URIs in one of the parents.

I think we need to do more to clean this up, but I didn't want to include too much test cleanup in this PR.